ergo/platforms/predictit.py

```diff
@@ -0,0 +1,332 @@
+    """
+        A PredictIt market.
+
+        :param predictit: PredictIt API instance
+        :param data: Market JSON retrieved from PredictIt API
+
+        :ivar PredictIt predictit: predictit api instance
+        :ivar str api_url: url of the predictit api for the given question
+        :ivar int id: id of the market
+        :ivar str name: name of the market
+        :ivar str shortName: shortened name of the market
+        :ivar str image: url of the image resource of the market
+        :ivar str url: url of the market in PredictIt
+        :ivar str status: status of the market. Closed markets aren't included in the API, so always "Open"
+        :ivar datetime.datetime timestamp: last time the market was updated.
+            Api updates every minute, but timestamp can be earlier if it hasn't been traded in
+    """
+
+    def __init__(self, predictit: Any, data: Dict):
+        self.predictit = predictit
+        self._load_attr(data)
+
+    def __repr__(self):
+        if self._data:
+            return f'<PredictItMarket title="{self.name}">'
+        else:
+            return "<PredictItMarket>"
+
+    def __str__(self):
+        return repr(self)
+
+    def _load_attr(self, data):
+        """
+        Load attributes of question from data to instance.
+        :param data:
+        """
+        self._data = data
+        self.id = data['id']
+        self.name = data['name']
+        self.shortName = data['shortName']
+        self.image = data['image']
+        self.url = data['url']
+        self.status = data['status']
+        self.api_url = f"{self.predictit.api_url}/markets/{self.id}/"
+        if data['timeStamp'] == 'N/A':
+            self.timestamp = None
+        else:
+            self.timestamp = parse(data['timeStamp'])
+
+    @property
+    def questions(self) -> Iterator[PredictItQuestion]:
+        """
+        Generate all of the questions in the market.
+        :return: iterator of questions in market
+        """
+        for data in self._data['contracts']:
+            yield PredictItQuestion(self, data)
+
+    def refresh(self):
+        """
+        Refetch the market data from PredictIt,
+        used when the question data might have changed.
+        """
+        r = self.predictit._get(self.api_url)
+        self._load_attr(r.json())
+
+    def get_question_by_id(self, id: int) -> PredictItQuestion:
+        """
+        Return the specified question given by the id number.
+        :param id:
+        :return: question
+        """
+        for question in self.questions:
+            if question.id == id:
+                return question
+        raise ValueError("Unable to find a question with that id.")
+
+    def search_question(self, name: str) -> PredictItQuestion:
+        """
+        Return the specified question given by the name of the question,
+        using fuzzy matching in the case where the name isn't exact.
+        :param name:
+        :return: question
+        """
```

Comment:
This is cool, but I'm not sure it should be core functionality. Could move it to `contrib/predictit/fuzzy_search` and make fuzzywuzzy an [optional dependency](https://github.com/oughtinc/ergo/blob/master/pyproject.toml#L24).